From fe17c9fadf375d8beeb42b062390dfe18cb59e08 Mon Sep 17 00:00:00 2001
From: Andreas Schneider <asn@cryptomilk.org>
Date: Fri, 11 Feb 2022 09:46:15 +0100
Subject: [PATCH 1/2] avcodec/exif: Include bytestream.h for GetByteContext

bytestream.h should be directly included for GetByteContext and not
rely on other headers to include it. It could be removed from there.

Signed-off-by: Andreas Schneider <asn@cryptomilk.org>
---
 libavcodec/exif.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libavcodec/exif.h b/libavcodec/exif.h
index 4db84a1c2f..310f6e8e64 100644
--- a/libavcodec/exif.h
+++ b/libavcodec/exif.h
@@ -30,6 +30,7 @@
 
 #include <stdint.h>
 #include "libavutil/dict.h"
+#include "bytestream.h"
 #include "tiff.h"
 
 #define EXIF_MAX_IFD_RECURSION 2
-- 
2.34.1


From 45a9f8044ef6dc425209141e3d9272008ce6bbff Mon Sep 17 00:00:00 2001
From: Andreas Schneider <asn@cryptomilk.org>
Date: Fri, 11 Feb 2022 09:49:25 +0100
Subject: [PATCH 2/2] avcodec/exif: Include tiff_common.h only where needed

The exif.h header doesn't use anything from tiff.h.

Signed-off-by: Andreas Schneider <asn@cryptomilk.org>
---
 libavcodec/exif.c     | 1 +
 libavcodec/exif.h     | 1 -
 libavcodec/mjpegdec.c | 1 +
 libavcodec/webp.c     | 1 +
 4 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/libavcodec/exif.c b/libavcodec/exif.c
index 0b656fd09b..9485b24b8e 100644
--- a/libavcodec/exif.c
+++ b/libavcodec/exif.c
@@ -26,6 +26,7 @@
  */
 
 #include "exif.h"
+#include "tiff_common.h"
 
 
 static const char *exif_get_tag_name(uint16_t id)
diff --git a/libavcodec/exif.h b/libavcodec/exif.h
index 310f6e8e64..ffacba7e66 100644
--- a/libavcodec/exif.h
+++ b/libavcodec/exif.h
@@ -31,7 +31,6 @@
 #include <stdint.h>
 #include "libavutil/dict.h"
 #include "bytestream.h"
-#include "tiff.h"
 
 #define EXIF_MAX_IFD_RECURSION 2
 #define EXIF_TAG_NAME_LENGTH   32
diff --git a/libavcodec/mjpegdec.c b/libavcodec/mjpegdec.c
index a735d2337d..267609d96a 100644
--- a/libavcodec/mjpegdec.c
+++ b/libavcodec/mjpegdec.c
@@ -50,6 +50,7 @@
 #include "tiff.h"
 #include "exif.h"
 #include "bytestream.h"
+#include "tiff_common.h"
 
 
 static int init_default_huffman_tables(MJpegDecodeContext *s)
diff --git a/libavcodec/webp.c b/libavcodec/webp.c
index 9e642e050a..148dc02170 100644
--- a/libavcodec/webp.c
+++ b/libavcodec/webp.c
@@ -48,6 +48,7 @@
 #include "exif.h"
 #include "get_bits.h"
 #include "internal.h"
+#include "tiff_common.h"
 #include "thread.h"
 #include "vp8.h"
 
-- 
2.34.1

